proxy_test.go

```diff
@@ -929,3 +930,93 @@ func TestGetSocketPath(t *testing.T) {
 	assert.NotNil(t, err, err)
 	assert.Equal(t, p, "")
 }
+
+func TestStoreRestore(t *testing.T) {
```

Comment:
Can you add tests for all the functions added? (This might necessitate splitting them into smaller functions to make them more easily testable).

Reply:
Ok, I'll try :)